contracts/BucketAuction.sol

```diff
@@ -95,6 +99,37 @@ contract BucketAuction is IBucketAuction, ERC721M {
         return _userData[user];
     }
 
+    function getUserDataPage(uint256 limit, uint256 offset)
+        external
+        view
+        returns (
+            User[] memory,
+            address[] memory,
+            uint256 total
+        )
+    {
+        uint256 numUsers = _users.length();
+        uint256 pageSize = limit;
```

Comment:
Do we need to cap `limit`? For example, maximum 100 etc.

Reply:
IMO its ok to not have limits here since the RPC node processing the request will just throw an error if the requested page limit is too expensive. Since the limit might be different across providers/depending on load, we can just make it configurable on the client side